### DISCUSSION TITLE
Issue when AOT enabled

### DISCUSSION THREAD

**Discussion by berik-99** [jchristn/WatsonTcp#307] [General]
I start by giving a big thank you to the creator of the library for all the hard work done so far and to all the contributors!

I tried to use the new AOT publishing feature, but I encountered some problems: 

### First Problem

The client execution stops with a NullReferenceException when calling the ``` Connect() ``` method, which obviously halts the execution.
the stack trace is: 

```
System.NullReferenceException: 'Object reference not set to an instance of an object.'
   at WatsonTcp.WatsonTcpClient.Connect()
   at Program.<Main>$(String[] args)
```

#### Steps to reproduce:
1. Create a Console Application project with AOT **disabled** and copy the server example from the README.
2. Create a second Console Application project with AOT **enabled**, and copy the client example from the README.
3. Start the server first, then the client.

### Second Problem

The server seems to work initially, but when a client connects, it is immediately disconnected without any direct exceptions on either side, although two exceptions are logged in the server application's debug console:

```
Exception thrown: 'System.InvalidOperationException' in System.Text.Json.dll
Exception thrown: 'System.InvalidOperationException' in System.Private.CoreLib.dll
```

#### Steps to reproduce:
1. Create a Console Application project with AOT **enabled** and copy the server example from the README.
2. Create a second Console Application project with AOT **disabled**, and copy the client example from the README.
3. Start the server first, then the client.

**Comment by anders9ustafsson** [comment]
Hi @berik-99 have you made any progress on the `NullReferenceException` issue? We are also occasionally getting this exception when trying to connect to a server that has recently been terminated. As far as I can see, it is not related to AOT enablement though.

**Comment by berik-99** [comment]
Hi @anders9ustafsson, honestly I haven't looked into the situation further as I've been busy with another project recently, but from today I can dedicate some time to this problem again. In the coming days, I'll try to analyze the situation more thoroughly to understand more. Any updates will be posted here.
